src/buffer.c

```diff
@@ -124,11 +124,26 @@ buffer_t *init_buffer(const char *filename) {
   // we want to start at the top when user opens up a new file
   buf->current = buf->head;
   // Set current to the first char of line
-  // TODO set current to the first non space char
   buf->current->current = buf->current->head;
 
-  buf->current_row = 0;
   buf->current_char = 0;
+  buf->current_row = 0;
+
+  // go to the first non-empty row
```

Comment:
I don't think Vim has such behaviour as it goes to the first non empty char but not first non empty line (if the first line is empty the cursor stays there)